nixos/modules/services/x11/desktop-managers/default.nix

```diff
@@ -73,7 +73,9 @@ in
             manage = "desktop";
             start = d.start
             + optionalString (needBGCond d) ''
-              if [ -e $HOME/.background-image ]; then
+              if [ -x $HOME/.background-image ]; then
```

Comment:
This will execute JPGs copied from a FAT filesystems, which often have the executable bit set in VFS due to permissive mount options. You could check for a `#!` or ELF magic bytes.

Reply:
I would much rather remove the entire section and let people set their own colours and wallpapers than try to work around every possible edge case here.